core/templates/dev/head/domain/exploration/QuestionListObjectFactory.js

```diff
@@ -35,6 +35,10 @@ oppia.factory('QuestionListObjectFactory', [function() {
     this._questions.push(question);
   };
 
+  QuestionList.prototype.updateQuestion = function(index, question) {
+    this._questions[index] = question;
```

Comment:
Just to check: is this intended to be a deep copy or not? I.e. if someone modifies the original question object should the changes be reflected here? If not you may want to do angular.copy() and in your jasmine test verify that the toEqual() check passes but the toBe() check doesn't.